bioblend/galaxy/histories/__init__.py

```diff
@@ -260,33 +337,25 @@ def download_dataset(self, history_id, dataset_id, file_path,
         """
         Download a ``dataset_id`` from history with ``history_id`` to a
         file on the local file system, saving it to ``file_path``.
-        """
-        # TODO: Outsource to DatasetClient.download_dataset() to replace most of this.
-        meta = self.show_dataset(history_id, dataset_id)
-        d_type = to_ext
-        if d_type is None and 'file_ext' in meta:
-            d_type = meta['file_ext']
-        elif d_type is None and 'data_type' in meta:
-            d_type = meta['data_type']
-
-        # TODO: Download this via the REST API. api/datasets/<dataset_id>/display
-        download_url = 'datasets/' + meta['id'] + '/display?to_ext=' + d_type
-        url = urlparse.urljoin(self.gi.base_url, download_url)
-
-        req = urllib2.urlopen(url)
-        if use_default_filename:
-            file_local_path = os.path.join(file_path, meta['name'])
-        else:
-            file_local_path = file_path
 
-        with open(file_local_path, 'wb') as fp:
-            shutil.copyfileobj(req, fp)
+        Legacy use only, refer to
+        ``bioblend.galaxy.dataset.DatasetClient.download_dataset()`` for
+        available parameters
```

Comment:
Hmm... structurally I love that this is no longer duplicated and is now doing the right thing. Not sure about calling this endpoint for legacy use only tough. I may prefer the idea of having two logical ways to download datasets (a history content centric way and a dataset centric way - depending on your current flow).

Thoughts?